pyqtgraph/parametertree/Parameter.py

```diff
@@ -466,11 +464,10 @@ def setLimits(self, limits):
         """Set limits on the acceptable values for this parameter. 
         The format of limits depends on the type of the parameter and
         some parameters do not make use of limits at all."""
-        if 'limits' in self.opts and fn.eq(self.opts['limits'], limits):
-            return
-        self.opts['limits'] = limits
-        self.sigLimitsChanged.emit(self, limits)
-        return limits
+        if not fn.eq(self.opts.get('limits'), limits):
```

Comment:
If `limits` *wasn't* in opts before and the new value is *None*, `sigLimitsChanged` won't end up firing because `self.opts.get('limits')` will return `None`. The old behavior should be able to catch this

Reply:
Ah, you're right! ... somehow expecting same handling of key `limits` missing and set to `None`, since this is what happens with `default` (`hasDefault() => False`). So this change would change behaviour, sorry...
Hmm, wouldn't it be better (clearer) to handle all options the same way? But this would surely require more changes throughout the code :-/. So, better revert this for now.